cmd/cvp-trigger/main.go

```diff
@@ -383,3 +387,17 @@ func appendMultiStageParams(podSpec *v1.PodSpec, params map[string]string) {
 		podSpec.Containers[0].Args = append(podSpec.Containers[0].Args, fmt.Sprintf("--multi-stage-param=%s=%s", key, params[key]))
 	}
 }
+
+func getProwWatcher(pjclient pjtypes.ProwJobInterface, selector fields.Selector) (watch.Interface, error) {
+	var w watch.Interface
+	var err error
+	err = wait.ExponentialBackoff(wait.Backoff{Steps: 10, Duration: 1 * time.Second, Factor: 2}, func() (bool, error) {
+		w, err = pjclient.Watch(context.TODO(), metav1.ListOptions{FieldSelector: selector.String()})
```

Comment:
While you assign to `err` here it will always be overwritten by the assignment from `wait.ExponentialBackoff()` - so it would be good to use a different variable here, as this is confusing as written.

Reply:
I know it predates this PR but `context.TODO()` is not production-ready code. I see you're importing the `interrupts` package - if you wait for graceful shutdown, you could use `interrupts.Context()` to cancel on process exit. Or, use `context.Background()` if you never want to handle cancellation, explicitly.

Reply:
Got it, thanks

Reply:
Aside from code clarity, is there any value with using `err` twice here?  The inner `err` is will be used on L396-397, and the outer `err` will be used in the return statement on L402.  I'm happy to change it for clarity, I just want to make sure there's no weird scoping behaviors in Go I'm unaware of.

I've fixed the `context.TODO()` and moved the code back into the `main()` function, but I wanted to ask the question before I push the updated commit and the diff changes.

Reply:
No, it should be functional with the way Go handles scoping/shadowing, just confusing.